python/beeai_framework/adapters/a2a/agents/agent.py

```diff
@@ -2,6 +2,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
 from collections.abc import AsyncGenerator
+from typing import Unpack
```

Comment:
`Unpack` needs to be imported from `typing_extensions` to work in Python before 3.13.
This should be caught by Ruff or Mypy. Would you mind taking a look to see if there is a rule that can check it?

Reply:
I checked and it's supported in standard `typing` from [3.11 onwards](https://docs.python.org/3/library/typing.html#typing.Unpack). This is why the linters are okay with it.
I also tested it on 3.11 on my machine, since it's the minimum we require in BeeAI.